pkg/virtual/framework/forwardingregistry/client.go

```diff
@@ -0,0 +1,210 @@
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	versionV1 = schema.GroupVersion{Version: "v1"}
+
+	deleteScheme          = runtime.NewScheme()
+	parameterScheme       = runtime.NewScheme()
+	deleteOptionsCodec    = serializer.NewCodecFactory(deleteScheme)
+	dynamicParameterCodec = runtime.NewParameterCodec(parameterScheme)
+)
+
+func init() {
+	metav1.AddToGroupVersion(parameterScheme, versionV1)
+	metav1.AddToGroupVersion(deleteScheme, versionV1)
+}
+
+type ResourceInterface interface {
+	dynamic.ResourceInterface
+	ResourceDeleterInterface
+}
+
+type ResourceDeleterInterface interface {
+	DeleteWithResult(ctx context.Context, name string, options metav1.DeleteOptions, subresources ...string) (*unstructured.Unstructured, int, error)
+	DeleteCollectionWithResult(ctx context.Context, options metav1.DeleteOptions, listOptions metav1.ListOptions) (*unstructured.Unstructured, error)
+}
+
+func NewClusterForConfig(c *rest.Config) (*Cluster, error) {
+	delegate, err := dynamic.NewClusterForConfig(c)
+	if err != nil {
+		return nil, err
+	}
+
+	config := dynamic.ConfigFor(c)
+	config.GroupVersion = &schema.GroupVersion{}
+	httpClient, err := rest.HTTPClientFor(config)
+	if err != nil {
+		return nil, err
+	}
+	restClient, err := rest.RESTClientForConfigAndClient(config, httpClient)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Cluster{delegate: delegate, scopedClient: &scopedClient{restClient}}, nil
+}
+
+type Cluster struct {
+	*scopedClient
+	delegate *dynamic.Cluster
+}
+
+func (c *Cluster) Cluster(name logicalcluster.Name) dynamic.Interface {
+	return &dynamicClient{
+		delegate:     c.delegate.Cluster(name),
+		scopedClient: c.scopedClient,
+		cluster:      name,
+	}
+}
+
+type dynamicClient struct {
+	*scopedClient
+	delegate dynamic.Interface
+	cluster  logicalcluster.Name
+}
+
+type scopedClient struct {
+	client *rest.RESTClient
+}
+
+var _ dynamic.Interface = &dynamicClient{}
+
+type dynamicResourceClient struct {
+	dynamic.ResourceInterface
+	delegate  dynamic.Interface
+	client    *rest.RESTClient
+	cluster   logicalcluster.Name
+	namespace string
+	resource  schema.GroupVersionResource
+}
+
+var _ ResourceInterface = &dynamicResourceClient{}
+
+func (c *dynamicClient) Resource(resource schema.GroupVersionResource) dynamic.NamespaceableResourceInterface {
+	return &dynamicResourceClient{
+		ResourceInterface: c.delegate.Resource(resource),
+		delegate:          c.delegate,
+		client:            c.client,
+		cluster:           c.cluster,
+		resource:          resource,
+	}
+}
+
+func (c *dynamicResourceClient) Namespace(ns string) dynamic.ResourceInterface {
+	ret := *c
+	ret.namespace = ns
+	ret.ResourceInterface = c.delegate.Resource(c.resource).Namespace(ns)
+	return &ret
+}
+
+func (c *dynamicResourceClient) DeleteWithResult(ctx context.Context, name string, options metav1.DeleteOptions, subresources ...string) (*unstructured.Unstructured, int, error) {
```

Comment:
can't we implement this is a function of a wrapper. Copying the whole dynamic client here is frightening.

Reply:
actually it's not the whole, is it?

Reply:
Right, it's only adding two methods for delete and deletecollection. For some reasons, the `dynamic.ResourceInterface` methods all return the REST request result in their signature, except these two.

Reply:
yes, got that. Move instantiation into the VW or even the registry and it's fine.